TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/BookActivity.kt

```diff
@@ -0,0 +1,258 @@
+                    if (book.currentChapter < book.numChapters - 1) {
+                        bookPagerView.isUserInputEnabled = false
+                        handler.postDelayed({
+                            book.currentChapter = book.currentChapter + 1
+                            book.currentPage = 0
+                            renderBook()
+                        }, 250)
+                    }
+                }
+                else -> {
+                    book.currentPage = if (layoutStateContainer.layoutMode == LayoutMode.NORMAL) {
+                        position - 1
+                    } else {
+                        (2 * (position - 1))
+                    }
+                }
+            }
+        }
+    }
+
+    inner class PageTransformer : ViewPager2.PageTransformer {
+        override fun transformPage(page: View, position: Float) {
+            page.apply {
+                when {
+                    position < -1 -> {
+                        translationX = 0f
+                        translationZ = 0f
+                        alpha = 0f
+                    }
+                    position > 1 -> {
+                        translationX = 0f
+                        translationZ = -1f
+                        alpha = 0f
+                    }
+                    position > 0 -> {
+                        translationX = width * -position
+                        translationZ = -1f
+                        alpha = 1f
+                    }
+                    else -> {
+                        translationX = 0f
+                        translationZ = 0f
+                        alpha = 1f
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.book_menu, menu)
+        val subMenu = menu?.findItem(R.id.action_chapter_group)?.subMenu
+        for (chapter in 0 until book.numChapters) {
+            subMenu?.add("Section $chapter")
+        }
+        subMenu?.add("Accreditation")
+        return true
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        onBackPressed()
+        return super.onSupportNavigateUp()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_font_up -> {
+                book.fontSize = min(24, book.fontSize + 2)
+                renderBook()
+            }
+            R.id.action_font_down -> {
+                book.fontSize = max(8, book.fontSize - 2)
+                renderBook()
+            }
+            else -> {
+                if (item.title == "Accreditation") {
+                    AlertDialog.Builder(this).setTitle("Project Gutenberg")
+                        .setMessage(this.resources.getString(R.string.accreditations))
+                        .show()
+                }
+                else if (item.title != null && "Section (\\d+)".toRegex().matches(item.title)) {
```

Comment:
the second part of the `if` statement could be extracted to its own function with a meaningful name so when you read the statement you understand it better:
```
else if (item.title != null && matchesWithTitle()) 
```

Reply:
using contains() instead of regex.matches() should be clearer